### PR TITLE
Fixes #33048 - Add Associate VM to dropdown

### PR DESCRIPTION
**Pull request by adiabramovitch**

### CODE REVIEW

**Review by yifatmakias — DISMISSED** [review]
@adiabramovitch nice work!
Looks good and works as expected.

File: app/helpers/compute_resources_vms_helper.rb [hunk 1]
```diff
@@ -263,23 +263,24 @@ def vm_import_action(vm, html_options = {})
...
-        :compute_resource_id => @compute_resource,
-        :id => vm.identity
-      ).merge(
-        :auth_object => @compute_resource,
-        :permission => 'edit_compute_resources'),
-      :title => _("Associate VM to a Foreman host"),
-      :method => :put,
-      :class => "btn btn-default"
-    )
+  def vm_associate_action(vm, isdropdown = false)
```
Comment (timogoebel) [inline app/helpers/compute_resources_vms_helper.rb]: Nit: The interface of this method is hard to use. `vm_associate_action(VM, false)` -> It's hard to figure out what `false` means.

Comment (adiabramovitch) [reply app/helpers/compute_resources_vms_helper.rb]: @timogoebel Thanks! do you have a better proposition? (:

Comment (timogoebel) [reply app/helpers/compute_resources_vms_helper.rb]: @adiabramovitch: Yep, I'd like to recommend this: https://thoughtbot.com/blog/ruby-2-keyword-arguments

File: app/helpers/compute_resources_vms_helper.rb [hunk 2]
```diff
@@ -263,23 +263,24 @@ def vm_import_action(vm, html_options = {})
...
-        :compute_resource_id => @compute_resource,
-        :id => vm.identity
-      ).merge(
-        :auth_object => @compute_resource,
-        :permission => 'edit_compute_resources'),
-      :title => _("Associate VM to a Foreman host"),
-      :method => :put,
-      :class => "btn btn-default"
-    )
+  def vm_associate_action(vm, isdropdown: false)
```
Comment (timogoebel) [inline app/helpers/compute_resources_vms_helper.rb]:
```suggestion
  def vm_associate_action(vm, is_dropdown: false)
```

Nit: This is how you'd name the variable in ruby.

File: app/helpers/compute_resources_vms_helper.rb [hunk 3]
```diff
@@ -263,23 +263,24 @@ def vm_import_action(vm, html_options = {})
...
+    if @compute_resource.supports_host_association?
```
Comment (timogoebel) [inline app/helpers/compute_resources_vms_helper.rb]:
```suggestion
  return unless @compute_resource.supports_host_association?
```

A guard clause can reduce the level of indentation.

File: app/helpers/compute_resources_vms_helper.rb [hunk 4]
```diff
@@ -263,23 +263,24 @@ def vm_import_action(vm, html_options = {})
...
+      display_link_if_authorized(
```
Comment (timogoebel) [inline app/helpers/compute_resources_vms_helper.rb]: Can you extract this to a helper method called `associate_vm_button(vm, klass)` and use this method in `vm_associate_action` and `ovirt_vms_data`?

I believe this helps a lot to declutter the code.
Sorry for being picky here, but this is not the prettiest code to begin with. I think we should make sure we clean it up a little when we have the chance.

Comment (adiabramovitch) [reply app/helpers/compute_resources_vms_helper.rb]: @timogoebel Can you clarify the extraction you are suggesting? I'm not sure I followed

Comment (timogoebel) [reply app/helpers/compute_resources_vms_helper.rb]: Something like this:

```ruby
def vm_associate_action(vm)
  associate_vm_link(vm, link_class: "btn btn-default")
end


def associate_vm_link(vm, link_class = '')
  display_link_if_authorized(
      _('Associate VM'),
      hash_for_associate_compute_resource_vm_path(
        :compute_resource_id => @compute_resource,
        :id => vm.identity
      ).merge(
        :auth_object => @compute_resource,
        :permission => 'edit_compute_resources'),
      :title => _("Associate VM to a Foreman host"),
      :method => :put,
      :class => link_class
    )
end

# and in ovirt_vms_data
vm_import_action(vm), vm_associate_link(vm),
```

**Comment by ShimShtein** [comment]
[test foreman]

File: ource.supports_host_association? [hunk 5]
```diff
@@ -0,0 +1,59 @@
+=>   <[1;34;4mFog[0m::[1;34;4mOvirt[0m::[1;34;4mCompute[0m::[1;34;4mServer[0m
```
Comment (ShimShtein) [inline ource.supports_host_association?]: I suppose this shouldn't be part of the commit

**Review by ShimShtein — CHANGES_REQUESTED** [review]
All in all the code looks good, but the rubocop error needs fixing and an extra file needs to be removed.

@adiabramovitch, how did you decide which resource types need the action and which do not? Or is it something that should be available in all other resource types too?

**Comment by yifatmakias** [comment]
[test katello]

**Review by ShimShtein — APPROVED** [review]
@timogoebel any concerns? I didn't test it, but the code looks good.

**Comment by timogoebel** [comment]
> @timogoebel any concerns?

No, I like how the code turned out. Imho it's a lot more readable this way. I didn't test this as well.

**Review by amirfefer — APPROVED** [review]
Tested with ec2,  a VM has been associated successfully
Thanks, @adiabramovitch!
